typegate/src/runtimes/patterns/messenger/async_messenger.ts

```diff
@@ -34,12 +42,48 @@ export class AsyncMessenger<Broker, M, A> {
     this.broker = broker;
     this.#send = send;
     this.#stop = stop;
+    this.initTimer();
   }
 
   async terminate(): Promise<void> {
     await Promise.all([...this.#tasks.values()].map((t) => t.promise));
     logger.info(`close worker ${this.constructor.name}`);
     this.#stop(this.broker);
+    clearInterval(this.#timer);
+  }
+
+  initTimer() {
+    if (this.#timer === undefined) {
+      this.#timer = setInterval(() => {
+        const currentDate = Date.now();
+        const item = this.#pendingOperations.peek(); // O(1)
+        if (item !== undefined) {
+          // safe removal
+          const delta = currentDate - item.date;
+          if (this.#doneIds.has(item.message.id)) {
+            this.#pendingOperations.pop(); // O(log N)
+            this.#doneIds.delete(item.message.id); // O(1)
+            return;
+          }
+
+          // force removal
+          const maxDurationMs = config.timer_max_timeout_ms;
+          if (delta >= maxDurationMs) {
+            this.receive({
+              id: item.message.id,
+              error: `${maxDurationMs / 1000}s timeout exceeded (+${
+                (delta - maxDurationMs) / 1000
+              }s)`,
+            });
+            this.#pendingOperations.pop(); // O(log N)
+
+            if (config.timer_destroy_ressources) {
+              this.#stop(this.broker); // force abort
```

Comment:
It should not only stop & cancel all ongoing promises (use a specific errors so the resolver throw an error explained it has been interrupted by another one), but also restart (if lazy easy, if sync a bit more complex) and make sure next requests are successfully fulfilled

Reply:
`restart` as in (1) restarting the task (restart broker + re-run task) or (2) restart the broker ?

Reply:
2 :)